src/movies.js

```diff
@@ -2,14 +2,137 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(array) {
+  let arrYear = array.slice();
+
+  arrYear.sort(function(a, b) {
+    if (a.year < b.year) {
+      return -1;
+    }
+    if (a.year > b.year) {
+      return 1;
+    }
+    if (a.year === b.year) {
+      if (a.title < b.title) {
+        return -1;
+      }
+      if (a.title > b.title) {
+        return 1;
+      }
+    }
+  });
+
+  return arrYear;
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(array) {
+  let moviesNum = 0;
+
+  const newArray = array.filter(whichMovie => {
+    let condition =
+      whichMovie.director === "Steven Spielberg" &&
+      whichMovie.genre.includes("Drama");
+
+    return condition;
+  });
+
+  moviesNum = newArray.length;
+
+  return moviesNum;
+}
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(array) {
+  let newArray = array.slice();
+
+  newArray.sort(function(a, b) {
+    if (a.title < b.title) {
+      return -1;
+    }
+    if (a.title > b.title) {
+      return 1;
+    }
+  });
+
+  const titleArr = newArray.map(function(movieTitle) {
```

Comment:
Performance wise, it would be better to do the map first to get the list of titles and then sort only the array of titles. That way the sort doesn't have to go over all of the objects inside the movies array.